saas/views/download.py

```diff
@@ -421,3 +424,273 @@ def queryrow_to_columns(self, record):
+                        **{heading: 0 for heading in self.headings if heading != 'Date'}
+                    }
+                consolidated_data[date_str][account['title']] = date_value[1]
+
+        return list(consolidated_data.values())
+
+    def queryrow_to_columns(self, record):
+        row = [
+            self.encode(record[heading]) for heading in self.headings
+        ]
+        return row
+
+
+class CustomerMetricsDownloadView(CSVDownloadView, CustomerMetricsMixin):
+    basename = 'customermetrics'
+
+    headings = [
+        'Date',
+        'Total # of Customers',
+        '# of new Customers',
+        '# of churned Customers',
+        'Net New Customers',
+        '% Customer Churn'
+    ]
+
+    def get_headings(self):
+        return self.headings
+
+    def get_queryset(self):
+        request = Request(self.request)
+        customer_table, _ = self.get_data(request=request)
+
+        consolidated_data = {}
+        for entry in customer_table:
+            for date_value in entry['values']:
+                date_str = date_value[0]
+                if date_str not in consolidated_data:
+                    consolidated_data[date_str] = {
+                        'Date': date_str,
+                        **{heading: 0 for heading in self.headings if heading != 'Date'}
+                    }
+                consolidated_data[date_str][entry['title']] = date_value[1]
+
+        return list(consolidated_data.values())
+
+    def queryrow_to_columns(self, record):
+        return [
+            self.encode(record[heading]) for heading in self.headings
+        ]
+
+
+class PlanMetricsDownloadView(CSVDownloadView, PlanMetricsMixin):
+    basename = 'planmetrics'
+
+    def set_plans_data(self, request):
+        plans, _ = self.get_data(request=Request(request))
+        self.plans = plans
+        self.headings = ['Date'] + [plan['title'] for plan in plans]
+
+    def get_headings(self):
+        return self.headings
+
+    def get_queryset(self):
+        csv_data = [
+            ['location'] + [plan['location'] for plan in self.plans],
+            ['is_active'] + ['true' if plan['is_active'] else 'false' for plan in self.plans]
+        ]
+
+        consolidated_data = {}
+        for plan in self.plans:
+            for date_value in plan['values']:
+                date_str = date_value[0]
+                value = date_value[1]
+                if date_str not in consolidated_data:
+                    consolidated_data[date_str] = [0] * len(self.plans)
+                index = [p['title'] for p in self.plans].index(plan['title'])
+                consolidated_data[date_str][index] = value
+
+        for date, values in consolidated_data.items():
+            row = [date] + values
+            csv_data.append(row)
+
+        return csv_data
+
+    def queryrow_to_columns(self, record):
+        return self.encode(record)
+
+    def get(self, request, *args, **kwargs):
+        self.set_plans_data(request)
```

Comment:
I don't understand the purpose of the complexity here. Is it to define `self.plans`? In that case why not use a property?

```
@property
def plans(self):
    if not hasattr(self, '_plans'):
        self._plans = ...
    return self._plans
```